File: packages/php/src/Types/Metadata/HeaderMetadata.php
<?php

declare(strict_types=1);

namespace Kreuzberg\Types\Metadata;

class HeaderMetadata
{
    public int $level;
    public string $text;
    public ?string $id = null;
    public int $depth;
    public int $htmlOffset;

    public function __construct(
        int $level,
        string $text,
        int $depth,
        int $htmlOffset,
        ?string $id = null,
    ) {
        $this->level = $level;
        $this->text = $text;
        $this->depth = $depth;
        $this->htmlOffset = $htmlOffset;
        $this->id = $id;
    }
}

File: packages/java/src/main/java/dev/kreuzberg/types/metadata/HeaderMetadata.java
package dev.kreuzberg.types.metadata;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.jspecify.annotations.Nullable;

/**
 * HTML header/heading metadata.
 */
public record HeaderMetadata(
    @JsonProperty("level") int level,
    @JsonProperty("text") String text,
    @JsonProperty("id") @Nullable String id,
    @JsonProperty("depth") int depth,
    @JsonProperty("html_offset") int htmlOffset
) { }
